cmd/insolard/main.go

```diff
@@ -10,55 +10,132 @@ import (
 	"io/ioutil"
 	"path/filepath"
 
-	"github.com/insolar/insolar/application"
-	appbuiltin "github.com/insolar/insolar/application/builtin"
-	"github.com/insolar/insolar/applicationbase/genesis"
-	"github.com/insolar/insolar/logicrunner/builtin"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 	jww "github.com/spf13/jwalterweatherman"
 
+	"github.com/insolar/insolar/application"
+	appbuiltin "github.com/insolar/insolar/application/builtin"
+	"github.com/insolar/insolar/applicationbase/genesis"
 	"github.com/insolar/insolar/certificate"
 	"github.com/insolar/insolar/configuration"
 	"github.com/insolar/insolar/insolar"
 	"github.com/insolar/insolar/log"
+	"github.com/insolar/insolar/logicrunner/builtin"
 	"github.com/insolar/insolar/server"
-	"github.com/insolar/insolar/version"
 )
 
 func main() {
 	var (
 		configPath        string
 		genesisConfigPath string
+		heavyDB           string
 		genesisOnly       bool
 	)
 
+	var cmdHeavy = &cobra.Command{
+		Use:   "heavy --config=path --heavy-genesis=path",
+		Short: "starts heavy node",
+		Run: func(cmd *cobra.Command, args []string) {
+			runHeavyNode(configPath, genesisConfigPath, heavyDB, genesisOnly)
+		},
+	}
+	cmdHeavy.Flags().StringVarP(&genesisConfigPath, "heavy-genesis", "", "", "path to genesis config for heavy node")
+	_ = cmdHeavy.MarkFlagRequired("heavy-genesis")
+	cmdHeavy.Flags().StringVarP(&heavyDB, "database", "", "", "sets database type for heavy node")
+	_ = cmdHeavy.MarkFlagRequired("database")
+	cmdHeavy.Flags().BoolVarP(&genesisOnly, "genesis-only", "", false, "run only genesis and then terminate")
+
+	var cmdLight = &cobra.Command{
+		Use:   "light --config=path",
+		Short: "starts light node",
+		Run: func(cmd *cobra.Command, args []string) {
+			runLightNode(configPath)
+		},
+	}
+
+	var cmdVirtual = &cobra.Command{
+		Use:   "virtual --config=path",
+		Short: "starts virtual node",
+		Run: func(cmd *cobra.Command, args []string) {
+			runVirtualNode(configPath)
+		},
+	}
+
 	var rootCmd = &cobra.Command{
 		Use: "insolard",
-		Run: func(_ *cobra.Command, _ []string) {
-			runInsolardServer(configPath, genesisConfigPath, genesisOnly)
-		},
 	}
-	rootCmd.Flags().StringVarP(&configPath, "config", "c", "", "path to config file")
-	rootCmd.Flags().StringVarP(&genesisConfigPath, "heavy-genesis", "", "", "path to genesis config for heavy node")
-	rootCmd.Flags().BoolVarP(&genesisOnly, "genesis-only", "", false, "run only genesis and then terminate")
-	rootCmd.AddCommand(version.GetCommand("insolard"))
+	rootCmd.PersistentFlags().StringVarP(&configPath, "config", "c", "", "path to config file")
+	_ = rootCmd.MarkPersistentFlagRequired("config")
```

Comment:
why are we skipping the error if the flag `config` does not exist?

Reply:
Thought it'll be obvious, I'll bring back errors